lib/Basics/files.cpp

```diff
@@ -2040,7 +2042,11 @@ static void SystemTempPathCleaner(void) {
   char* path = SystemTempPath.get();
 
   if (path != nullptr) {
-    TRI_RMDIR(path);
+    namespace fs=boost::filesystem;
+    fs::path pathToRemove(path);
+    for (fs::directory_iterator end, it(pathToRemove); it!=end; ++it) {
```

Comment:
`TRI_RMDIR(path)` will on Linux and MacOS call `rmdir (2)`, which will remove an empty directory.
That means the prerequisite for this call is that the directory must have been empty already. Otherwise `rmdir (2)` does not succeed.
The new implementation assumes the directory to be removed is NOT empty, because it will iterate over all the files in it...
Additionally, TRI_RMDIR was exception-safe, and the new version is not. Too bad it is called on shutdown.